src/vs/workbench/browser/parts/titlebar/titlebarPart.ts

```diff
@@ -182,10 +222,33 @@ export class BrowserTitleService extends MultiWindowParts<BrowserTitlebarPart> i
 	private variables: ITitleVariable[] = [];
 
 	registerVariables(variables: ITitleVariable[]): void {
-		this.variables.push(...variables);
+		const unregisteredVariables: ITitleVariable[] = [];
+
+		// Filter out any variables that are already registered
+		const existingVariableNames = new Set(this.variables.map(v => v.name));
+		const existingVariableContexKeys = new Set(this.variables.map(v => v.contextKey));
+		for (const v of variables) {
+			if (existingVariableNames.has(v.name)) {
+				console.error(`A title variable with the name '${v.name}' is already registered.`);
+			} else if (existingVariableContexKeys.has(v.contextKey)) {
+				console.error(`A title variable with the context key '${v.contextKey}' is already registered.`);
+			} else {
+				unregisteredVariables.push(v);
+			}
+		}
+
+		this.variables.push(...unregisteredVariables);
 
 		for (const part of this.parts) {
-			part.registerVariables(variables);
+			part.registerVariables(unregisteredVariables);
+		}
+	}
+
+	unregisterVariables(variableNames: string[]): void {
+		this.variables = this.variables.filter(v => !variableNames.includes(v.name));
```

Comment:
I think either we can return a `IDisposable` or we should not have this method. I find it strange that I can unregister any kind of variable I did not create.